gordon/metrics/log.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+import logging
+import time
+
+import zope.interface
+
+from gordon import interfaces
+
+# TODO: update plugin loader to load metrics logger as default
+
+
+@zope.interface.implementer(interfaces.ITimer)
+class LogTimer:
+    """Timer that outputs metrics to the logger.
+
+    Args:
+        metric (dict): Metric to output.
+        logger (.MetricLogger): Object used for outputting metrics.
+        time_unit (number): (optional) Multiplier used to scale time.time()
+            output. For example, a value of `1` outputs whole seconds,
+            a value of `1000` outputs milliseconds.
+    """
+    def __init__(self, metric, logger, time_unit):
+        self.metric = metric
+        self.logger = logger
+        self.time_unit = time_unit
+
+    async def __aenter__(self):
+        await self.start()
+        return self
+
+    async def __aexit__(self, *args):
+        await self.stop()
+
+    async def start(self):
+        """Start the timer."""
+        self._start_time = time.time()
```

Comment:
So I recently learned that `time.time` is not guaranteed to be monotonic (ntp / clock adjustments happen often, affected by admin adjustments, etc). It'd suggest `time.perf_counter` (it's CPU time, which when we want to measure how long some logic takes, makes sense, since that differs across machines), or `time.monotonic` (less precise than perf_counter). [A good post on the differences](https://www.webucator.com/blog/2015/08/python-clocks-explained/) (disclaimer: I've only skimmed).